frontend/src/Menu/index.jsx

```diff
@@ -9,7 +11,8 @@ import leaderBoardMenuIcon from '../../assets/menu/Leaderboard.png';
 const HEIGHT_RATIO = 0.25;
 
 function Menu() {
-  const [isPressed, setPressed] = useState('Buildings');
+  const isPressed = useSelector((state) => state.menu);
```

Comment:
from the 'isPressed' I would expect a boolean value.
please rename to something descriptive (currentlyRenderedComponent, ...)
rename state.menu, to state.currentlySelectedMenuItem or something

Reply:
if it's a function, activatingMenu is also fine, but if I'm correct here isPressed is supposed to be the component, like 'Buildings', right?